pupil_src/shared_modules/marker_tracker_3d/optimization/model_state.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import os
+
+import numpy as np
+
+import file_methods
+from marker_tracker_3d import utils
+
+logger = logging.getLogger(__name__)
+
+
+class ModelState:
+    def __init__(self, save_path):
+        self.save_path = save_path
+
+        self.current_frame_id = 0
+
+        self.frames_id = []
+        self.markers_id = []
+
+        self.all_novel_markers = []
+
+        self.camera_extrinsics_opt = {}  # {frame id: camera extrinsics(list)}
+        self.marker_extrinsics_opt = {}  # {marker id: marker extrinsics(list)}
+        self.marker_points_3d_opt = {}
+
+        self.model = None
+        self.load_marker_tracker_3d_model_from_file()
+
+    def load_marker_tracker_3d_model_from_file(self):
+        self.model = file_methods.Persistent_Dict(
+            os.path.join(self.save_path, "marker_tracker_3d_model")
+        )
+        self.marker_extrinsics_opt = {
+            k: np.array(v)
+            for k, v in self.model.get("marker_extrinsics_opt", {}).items()
```

Comment:
I would prefer more explicit variable names. If you want to keep them very short, `id` and `extr` are fine, but `marker_id` and `extrinsics` wouldn't hurt either.
The same applies to the other dict iterations in the file.